### PR TITLE
Homework 10, Lesson 12

### PR DESCRIPTION
**Pull request by AlkMkr**

### CODE REVIEW

File: Oleksii_Makarov/HW12/Lesson12_1.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            Returns the number of employees
+        """
+        return self.__employees
+
+    @employees.setter
+    def employees(self, new_value):
+        """
+            Replaces the number of employees
+        """
+        if type(new_value) == int:
```
Comment (bohdandovbsyh) [inline Oleksii_Makarov/HW12/Lesson12_1.py]: -1000 is a integer

File: Oleksii_Makarov/HW12/Lesson12_1.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            self.__employees = new_value
+        else:
+            raise TypeError("Number of employees must be an int")
+
+    def employees_plus(self, additional):
+        """
+            Function describing addition of number of employees
+        """
+        if type(additional) == int:
```
Comment (bohdandovbsyh) [inline Oleksii_Makarov/HW12/Lesson12_1.py]: -100000 is an integer

File: Oleksii_Makarov/HW12/Lesson12_1.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            Returns the location of company's Headquarters
+        """
+        return self.__hq_location
+
+    @hq_location.setter
+    def hq_location(self, new_location):
+        """
+            Set a new location for a company's HQ if it's string
+        """
+        if type(new_location) == str:
```
Comment (bohdandovbsyh) [inline Oleksii_Makarov/HW12/Lesson12_1.py]: '' it is a string

File: Oleksii_Makarov/HW12/Lesson12_1.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        else:
+            raise TypeError("The location must be a string")
+
+    @hq_location.deleter
+    def hq_location(self):
+        """
+            Removes company's headquarters location. Boom.
+        """
+        self.__hq_location = None
+
```
Comment (bohdandovbsyh) [inline Oleksii_Makarov/HW12/Lesson12_1.py:102]: please add general methods. not only properties

File: Oleksii_Makarov/HW12/Lesson12_2.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+            Returns age of employee in full years
+        """
+        return self.__age
+
+    @age.setter
+    def age(self, new_age):
+        """
+            Set an employee's age to a new full number. Age must be an int
+        """
+        if type(new_age) == int:
```
Comment (bohdandovbsyh) [inline Oleksii_Makarov/HW12/Lesson12_2.py]: -100 is integer

File: Oleksii_Makarov/HW12/Lesson12_2.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+            Returns a position of employee
+        """
+        return self.__position
+
+    @position.setter
+    def position(self, new_position):
+        """
+            Set a new position for employee. Position should be a string
+        """
+        if type(new_position) == str:
```
Comment (bohdandovbsyh) [inline Oleksii_Makarov/HW12/Lesson12_2.py]: '' it is a string as well

File: Oleksii_Makarov/HW12/Lesson12_2.py [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+            Returns salary of employee in USD
+        """
+        return f'{self.__salary}$'
+
+    @salary.setter
+    def salary(self, new_salary):
+        """
+            Sets salary of the employee. Must be a float
+        """
+        if type(new_salary) == float:
```
Comment (bohdandovbsyh) [inline Oleksii_Makarov/HW12/Lesson12_2.py]: -100000.0 is a float

File: Oleksii_Makarov/HW12/Lesson12_2.py [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+        """
+        if type(new_experience) == float:
+            if new_experience <= self.__experience:
+                raise ValueError("New experience must be greater than old one")
+            else:
+                self.__experience = new_experience
+        else:
+            raise TypeError("setter accepts only float")
+
+    def experience_adder(self, additional_exp: float):
```
Comment (bohdandovbsyh) [inline Oleksii_Makarov/HW12/Lesson12_2.py]: what is the difference between setter and this function?

File: Oleksii_Makarov/HW12/Lesson12_1.py (hunk 1 above)
Comment (AlkMkr) [reply Oleksii_Makarov/HW12/Lesson12_1.py]: Fixed. Check

**Review by bohdandovbsyh — APPROVED** [review]

**Pull request merged**